my_2d_robot/src/my_2d_robo_state_publisher.cpp

```diff
@@ -0,0 +1,42 @@
+#include <string>
+#include <cstdlib>
+#include <ctime>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include "std_msgs/String.h"
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "my_2d_robo_state_publisher");
+  ros::NodeHandle n;
+  ros::Publisher joint_pub = n.advertise<sensor_msgs::JointState>("joint_states", 1000);
+  ros::Rate loop_rate(30);
+
+  sensor_msgs::JointState joint_state;
+
+  srand(time(NULL));
+
+  while (ros::ok)
+  {
+    joint_state.header.stamp = ros::Time::now();
+    // There is one message per update. This is 2 since 2 joints are updated at a time.
+    joint_state.name = { "joint1", "joint2" };
+
+    float joint1_pos = ((rand() % 315) / 100);
+    float joint2_pos = ((rand() % 315) / 100);
+
+    // The position is the angle of rotation.
+    joint_state.position = { joint1_pos, joint2_pos };
```

Comment:
The visualization should show both arms moving, but only one does.

![only_one_arm_updates](https://user-images.githubusercontent.com/35084180/88408121-1bea7380-cda1-11ea-81da-80a60fc31e9f.gif)

Reply:
I think I've solved this